addons/mail/static/tests/discuss_app/im_status.test.js

```diff
@@ -104,3 +109,47 @@ test("show im status in messaging menu preview of chat", async () => {
         contains: ["i[aria-label='User is online']"],
     });
 });
+
+test("subscribe to im status when member joined the channel", async () => {
+    const pyEnv = await startServer();
+    const demoPartner = pyEnv["res.partner"].create({ name: "Demo" });
+    pyEnv["res.users"].create({ partner_id: demoPartner });
+    const [generalId] = pyEnv["discuss.channel"].create([
+        { name: "General" },
+        {
+            name: "Sales",
+            channel_member_ids: [],
+        },
+    ]);
+    // Ensure that the general channel does not interfere with subscriptions (we
+    // subscribe when self member joined after the bus started).
+    mockDate("2024-08-02");
+    const env = await start();
+    patchWithCleanup(env.services.bus_service, {
+        forceUpdateChannels: () => step("update-channels"),
+    });
+    await openDiscuss(generalId);
+    await contains(".o-mail-Discuss-threadName[title=General]");
+    await assertSteps([]);
+    await click(".o-mail-Discuss-header button[title='Add Users']");
+    await click(".o-discuss-ChannelInvitation-selectable");
```

Comment:
Keep brief selectors:
- `.o-mail-Discuss-header` not really useful (even `button` could be dropped)
- `.o-discuss-ChannelInvitation-selectable` is not clear enough in functional intent. This is more a style/internal component classname. For test that should select a user to invite, explicitly click on checkbox with the given name.